examples/fillstates.py

```diff
@@ -80,18 +89,59 @@
-# cycle through state names, color each one.
+
+#%% ---------  cycle through state names, color each one.  --------------------
+fig = plt.gcf()  # get current figure instance
 ax = plt.gca() # get current axes instance
+
 for nshape,seg in enumerate(m.states):
     # skip DC and Puerto Rico.
-    if statenames[nshape] not in ['District of Columbia','Puerto Rico']:
-        color = rgb2hex(colors[statenames[nshape]]) 
+    if statenames[nshape] not in ['Puerto Rico', 'District of Columbia']:
+        color = rgb2hex(colors[statenames[nshape]])
         poly = Polygon(seg,facecolor=color,edgecolor=color)
         ax.add_patch(poly)
-# draw meridians and parallels.
-m.drawparallels(np.arange(25,65,20),labels=[1,0,0,0])
-m.drawmeridians(np.arange(-120,-40,20),labels=[0,0,0,1])
-plt.title('Filling State Polygons by Population Density')
+
+AREA_1 = 0.005  # exclude small Hawaiian islands that are smaller than AREA_1
+AREA_2 = AREA_1 * 30.0  # exclude Alaskan islands that are smaller than AREA_2
+AK_SCALE = 0.18  # scale down Alaska to show as a map inset
+
+for nshape, shapedict in enumerate(m_.states_info):  # plot Alaska and Hawaii as map insets
+    if shapedict['NAME'] in ['Alaska', 'Hawaii']:
+        seg = m_.states[int(shapedict['SHAPENUM'] - 1)]
+        if shapedict['NAME'] == 'Hawaii' and float(shapedict['AREA']) > AREA_1:
+            seg = list(map(lambda (x,y): (x-1900000, y+250000), seg))
+            color = rgb2hex(colors[statenames[nshape]])
+        elif shapedict['NAME'] == 'Alaska' and float(shapedict['AREA']) > AREA_2:
+            seg = list(map(lambda (x,y): (AK_SCALE*x-200000, AK_SCALE*y-650000), seg))
+            color = rgb2hex(colors[statenames[nshape]])
+        poly = Polygon(seg, facecolor=color, edgecolor='gray', linewidth=.45)
+        ax.add_patch(poly)
+
+ax.set_title('United states population density by state')
+
+#%% ---------  Plot bounding boxes for Alaska and Hawaii insets  --------------
+light_gray = [0.8]*3  # define light gray color RGB
+m_.plot(np.linspace(170,177),np.linspace(29,29),linewidth=1.,
+        color=light_gray,latlon=True)
+m_.plot(np.linspace(177,180),np.linspace(29,26),linewidth=1.,
+        color=light_gray,latlon=True)
+m_.plot(np.linspace(180,180),np.linspace(26,23),linewidth=1.,
+        color=light_gray,latlon=True)
+m_.plot(np.linspace(-180,-177),np.linspace(23,20),linewidth=1.,
+        color=light_gray,latlon=True)
+m_.plot(np.linspace(-180,-175),np.linspace(26,26),linewidth=1.,
+        color=light_gray,latlon=True)
+m_.plot(np.linspace(-175,-171),np.linspace(26,22),linewidth=1.,
+        color=light_gray,latlon=True)
+m_.plot(np.linspace(-171,-171),np.linspace(22,20),linewidth=1.,
+        color=light_gray,latlon=True)
+
+#%% ---------   Show color bar  ---------------------------------------
+ax_c = fig.add_axes([0.9, 0.1, 0.03, 0.8])
+cb = ColorbarBase(ax_c,cmap=cmap,norm=norm,orientation='vertical',
+                  label=r'[population per $\mathregular{km^2}$]')
```

Comment:
Why not a regular colorbar creation approach instead of manually creating axes and such? An example like this is already getting pretty complicated, so we should aim to reduce complexity where-ever possible.

Reply:
Hi @jsh9:

I have this file around where I was coloring countries based on population. I used it for teaching purposes, so there is an intentional bug somewhere, but you can check out how I did the colorbar there at the end:

https://github.com/guziy/CMOS-python-tutorial/blob/master/fiona_test.py

Cheers